Source/JavaScriptCore/b3/testb3_2.cpp

```diff
@@ -4899,100 +4897,96 @@ void testExtractRegister32()
         Value* mValue = root->appendNew<Value>(
             proc, Trunc, Origin(), 
             root->appendNew<ArgumentRegValue>(proc, Origin(), GPRInfo::argumentGPR1));
-        Value* mask1Value = root->appendNew<Const32Value>(proc, Origin(), mask1);
-        Value* mask2Value = root->appendNew<Const32Value>(proc, Origin(), mask2);
+        Value* maskValue = root->appendNew<Const32Value>(proc, Origin(), mask);
         Value* highWidthValue = root->appendNew<Const32Value>(proc, Origin(), highWidth);
         Value* lowWidthValue = root->appendNew<Const32Value>(proc, Origin(), lowWidth);
 
-        Value* leftAndValue = root->appendNew<Value>(proc, BitAnd, Origin(), nValue, mask1Value);
+        Value* leftAndValue = root->appendNew<Value>(proc, BitAnd, Origin(), nValue, maskValue);
         Value* left = root->appendNew<Value>(proc, Shl, Origin(), leftAndValue, highWidthValue);
-
-        Value* rightAndValue = root->appendNew<Value>(proc, BitAnd, Origin(), mValue, mask2Value);
-        Value* right = root->appendNew<Value>(proc, ZShr, Origin(), rightAndValue, lowWidthValue);
+        Value* right = root->appendNew<Value>(proc, ZShr, Origin(), mValue, lowWidthValue);
 
         root->appendNewControlValue(
             proc, Return, Origin(), 
             root->appendNew<Value>(proc, BitOr, Origin(), left, right));
 
         auto code = compileProc(proc);
-        if (isARM64() && lowWidth > 0)
+        if (checkEmittedEXTR && isARM64() && 0 < lowWidth && lowWidth < 32)
```

Comment:
Why do we have `checkEmittedEXTR`? Shouldn't we always be checking for the instruction when we think we should have it?

Reply:
Some certain `lowWidth` values can trigger other optimizations in `B3ReduceStrength.cpp`, which can break this `EXTR` pattern. Then, `checkEmittedEXTR` is introduced since not every test case can emit `EXTR`.